In [1]:
from sklearn.datasets import load_files
import pandas as pd
import numpy as np

In [2]:
#Malayalam
Mal_train=pd.read_csv("tamil_sentiment_full_train.tsv", sep='\t')


Mal_Dev=pd.read_csv("tamil_sentiment_full_dev.tsv", sep='\t')


Mal_test=pd.read_csv("tamil_sentiment_full_test_withoutlabels.tsv", sep='\t')

In [3]:
Mal_train.shape, Mal_Dev.shape, Mal_test.shape

((35656, 2), (3962, 2), (4402, 2))

In [4]:
Mal_train

text        category
0      Vani bhojam fans hit like solli 500 like Vangi...   unknown_state
1                           I love you ajith very I like        Positive
2      ennaya trailer Ku mudi Ellam nikkudhu... Vera ...        Positive
3               Vijay Annaa  Ur Maassssss Therrrrriiiiii        Positive
4                            நம்ப நடே நாசாமா தான் போச்சி        Negative
...                                                  ...             ...
35651  ஒருவர் செய்த தவறுக்காக எல்லாரையும் பழி போடுவது...  Mixed_feelings
35652  Repeated Mode to Watching Theri Trailer & Ther...        Positive
35653  yeevanikuachu  pula kutti erudhal oodi poyidu....        Negative
35654                 She is looking like laughing budha        Positive
35655                   NORTH OR SOUTH KGF IS THE BEST!!        Positive

[35656 rows x 2 columns]

In [5]:
Mal_train['category']=Mal_train['category'].astype(str)
labels,counts=np.unique(Mal_train['category'],return_counts=True)
print(labels,counts)

['Mixed_feelings' 'Negative' 'Positive' 'Positive ' 'not-Tamil'
 'unknown_state'] [ 4020  4271 20069     1  1667  5628]


In [6]:
Mal_test.head()

id                                               text
0  Tam_1  வீர செங்குந்தர் சார்பாக இந்த திரைப்படம் வெற்றி...
1  Tam_2  Teruk ah irukku .... mokke movie .. waste of time
2  Tam_3  manitha samuthaayam amaipil irunthu intha pada...
3  Tam_4                                   JJ mam we miss u
4  Tam_5          Subtitle me traller dekhne wale like karo

In [7]:
#from ml2en import ml2en

In [8]:
#converter = ml2en()

#desti_lang={
 #    'malayalam' : 'ml'
#}

In [9]:
#for key,value in desti_lang.items():
     #print(transt.translate(Mal_train['text']))
       # Mal_train['text1']=Mal_train['text'].apply(lambda x: converter.transliterate(x))
     #   Mal_Dev['text1']=Mal_Dev['text'].apply(lambda x: converter.transliterate(x))
      #  Mal_test['text1']=Mal_test['text'].apply(lambda x: converter.transliterate(x))#

In [10]:
Mal_train[61:70]

text        category
61       Today trailer paaka yaarellam wait panreenga   unknown_state
62  mayiru maadiri iruku thevidiya pasagala Rajini...        Negative
63  Saguni padam madhiri theriudhu,soori irukkurad...  Mixed_feelings
64  Ada paavingala mangatha theme. Aaa. Slow. Moti...        Negative
65  Kangana look like same I just love this first ...        Positive
66  wowwwwwwwwww thalaya aaaaaa masssssssss dialog...        Positive
67  3 supar dupar star akshaya rajnikant and any j...        Positive
68                  Coffee solunga marana masss thala        Positive
69                            598K left for 15M views   unknown_state

In [11]:
Mal_Dev.head()

text       category
0  @0:40 songsuperb..kamallllla..  I'm waiting fo...  unknown_state
1  கணத்ததோர் அகமுடையார் சார்பாக  படம் வெற்றி அடைய...       Positive
2  Thalavia neenga veera level boss and neega tha...       Positive
3              Oru padam patha fell.vera level music       Positive
4  Hairstyle than mattama iruku. Adhu mattum math...  unknown_state

In [12]:
Mal_train=Mal_train.replace(to_replace =["Positive "], value ="Positive") 
Mal_train.shape

(35656, 2)

In [13]:
# Train
Mal_train['category'].value_counts()

Positive          20070
unknown_state      5628
Negative           4271
Mixed_feelings     4020
not-Tamil          1667
Name: category, dtype: int64

In [14]:
# Development 
Mal_Dev['category'].value_counts()

Positive          2257
unknown_state      611
Negative           480
Mixed_feelings     438
not-Tamil          176
Name: category, dtype: int64

In [15]:
msg=Mal_train['text']

msgDev=Mal_Dev['text']

msgTest=Mal_test['text']

In [16]:
msgDev[50:70]

50    Bigil=like Kaithi= cmnt Idhu epdi irruku vj fn...
51    Camne bole trending kt malaysia? Yg dri malays...
52             Auto like alludhu vekkama illayada?? Thu
53    Itha paartha kaapan maathiri illa adhavan 2 ma...
54                         333k likes thala Ajith storm
55    மோகன் அண்ணா மேன்மேலும் திரைப்படத்துறையில் வளர ...
56    01.Namba kita kaadu iruntha eduthukuvango 02.r...
57    சாதிய ஒழிக்கனுனு koovitu... சாதி வெறி புடிச்ச ...
58    I think sema line Ravi pidisu irukaaru. Sema j...
59    Like here like here nu solravanga enaku like p...
60    அடுத்தவங்க இசையை பயன்படுத்துகிறவங்க எல்லாம்......
61    My fevret Hero Akshay kumar  Supar Dupar hogi ...
62    VANNIYAR kottai dharmapuri padaiyachi Vanniyar...
63                    Pwoli  rakshit shetty kerala fans
64    Simbu zindhabad from all kannada power  puneet...
65    Avalavu nalla illai.. I'm not a superstar hate...
66                          Str da ..goyala str fans da
67           Padoo mass ah iruka podhu but tn la

In [17]:
#Remove Emojis

def emoji(msg): 
    
    msg_emoj=msg.str.replace(r'[(\U0001F600-\U0001F92F|\U0001F300-\U0001F5FF|\U0001F680-\U0001F6FF|\U0001F190-\U0001F1FF|\U00002702-\U000027B0|\U0001F926-\U0001FA9F|\u200d|\u2640-\u2642|\u2600-\u2B55|\u23cf|\u23e9|\u231a|\ufe0f)]+','')
    
    msg_digit=msg_emoj.str.replace(r'[0-9]', ' ')

    msg_Spac = msg_digit.str.replace(r'_',' ')

    msg_Spac = msg_Spac.str.replace(r'.','')

    msg_Spac = msg_Spac.str.replace(r'!','')

    msg_Spac = msg_Spac.str.replace(r'#','')

    msg_Spac = msg_Spac.str.replace(r'%','')

    msg_Spac = msg_Spac.str.replace(r'&','')

    msg_Spac = msg_Spac.str.replace(r'’','')

    msg_Spac = msg_Spac.str.replace(r'(','')

    msg_Spac = msg_Spac.str.replace(r')','')

    msg_Spac = msg_Spac.str.replace(r'-','')

    msg_Spac = msg_Spac.str.replace(r'/','')

    msg_Spac = msg_Spac.str.replace(r':','')

    msg_Spac = msg_Spac.str.replace(r';','')

    msg_Spac = msg_Spac.str.replace(r'<','')

    msg_Spac = msg_Spac.str.replace(r'=','')

    msg_Spac = msg_Spac.str.replace(r'>','')

    msg_Spac = msg_Spac.str.replace(r'?','')

    msg_Spac = msg_Spac.str.replace(r'@','')

    msg_Spac = msg_Spac.str.replace(r'[','')

    msg_Spac = msg_Spac.str.replace(r']','')

    msg_Spac = msg_Spac.str.replace(r'^','')

    msg_Spac = msg_Spac.str.replace(r'{','')

    msg_Spac = msg_Spac.str.replace(r'}','')

    msg_Spac = msg_Spac.str.replace(r'|','')

    msg_Spac = msg_Spac.str.replace(r'~','')

    msg_Spac = msg_Spac.str.replace(r'+','')

    msg_Spac = msg_Spac.str.replace(r'\s+', ' ')


    return msg_Spac
    

In [18]:
msg_Train=emoji(msg)

msg_Dev=emoji(msgDev)

msg_Test=emoji(msgTest)

e:\Anaconda3\lib\site-packages\ipykernel_launcher.py:5: FutureWarning: The default value of regex will change from True to False in a future version.
  """
e:\Anaconda3\lib\site-packages\ipykernel_launcher.py:7: FutureWarning: The default value of regex will change from True to False in a future version.
  import sys
e:\Anaconda3\lib\site-packages\ipykernel_launcher.py:11: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  # This is added back by InteractiveShellApp.init_path()
e:\Anaconda3\lib\site-packages\ipykernel_launcher.py:23: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
e:\Anaconda3\lib\site-packages\ipykernel_launcher.py:25: FutureWarning: The default value of regex will c

In [19]:
msg_Test.head()

0    வீர செங்குந்தர் சார்பாக இந்த திரைப்படம் வெற்றி...
1            Teruk ah irukku mokke movie waste of time
2    manitha samuthaayam amaipil irunthu intha pada...
3                                     JJ mam we miss u
4            Subtitle me traller dekhne wale like karo
Name: text, dtype: object

In [20]:
msg_Train[78:90], msg_Dev[50:70], msg_Test[21:22]


(78    Cmnt section pona like here like here nu uyira...
 79                        tranding kyu ni aa rahi h yrr
 80                  Mrana Mass iliya suppr star DHANUSH
 81    vera level pakka blockbuster for sure all d be...
 82    வன்னியர் சாதி சார்பாக படம் வெற்றி பெற வாழ்த்து...
 83             Yuvan bgm vera level i love u yuvan anna
 84    Rishi Richard Fans Club சார்பாக படம் வெற்றியடை...
 85    ஜெயம் ரவி அண்ணா வாழ்த்துக்கள் நீங்க இதே மாதிரி...
 86    படம் வெற்றிகரமாக அமைய வாழ்த்துக்கள் செளராஷ்டிர...
 87                      what is this rubbish Ayoooo ayo
 88            தரமா டீஸர் அதேபோல் படமும் தரமாக இருக்கும்
 89    Maximum intha mathiri msg kudukara reality Fil...
 Name: text, dtype: object,
 50    Bigillike Kaithi cmnt Idhu epdi irruku vj fns ...
 51    Camne bole trending kt malaysia Yg dri malaysi...
 52               Auto like alludhu vekkama illayada Thu
 53    Itha paartha kaapan maathiri illa adhavan maat...
 54                            k likes thala Ajith storm
 55

In [21]:
Train_Df=pd.DataFrame(msg_Train)
Train_Df.columns=['Msg_with_Stopwords']

Dev_Df=pd.DataFrame(msg_Dev)
Dev_Df.columns=['Msg_with_Stopwords']

Test_Df=pd.DataFrame(msg_Test)
Test_Df.columns=['Msg_with_Stopwords']

In [22]:
Train_Df.shape, Dev_Df.shape, Test_Df.shape

((35656, 1), (3962, 1), (4402, 1))

In [23]:
# Removing Null Values

Train_Df['Msg_with_Stopwords']=Train_Df['Msg_with_Stopwords'].fillna("")

Dev_Df['Msg_with_Stopwords']=Dev_Df['Msg_with_Stopwords'].fillna("")

Test_Df['Msg_with_Stopwords']=Test_Df['Msg_with_Stopwords'].fillna("")


In [24]:
Mal_train.shape, Mal_Dev.shape,Train_Df.shape, Dev_Df.shape, Mal_test.shape, Test_Df.shape

((35656, 2), (3962, 2), (35656, 1), (3962, 1), (4402, 2), (4402, 1))

In [25]:
import nltk
nltk.download('stopwords')
from nltk.corpus import stopwords
from nltk.tokenize import word_tokenize

[nltk_data] Error loading stopwords: <urlopen error [Errno 11001]
[nltk_data]     getaddrinfo failed>


In [26]:
stop_words = set(stopwords.words('english'))

In [27]:
Train_Df['Msg_without_Stopwords']=Train_Df['Msg_with_Stopwords'].apply(lambda x: ' '.join([word for word in x.split() if word not in (stop_words)]))

Dev_Df['Msg_without_Stopwords']=Dev_Df['Msg_with_Stopwords'].apply(lambda x: ' '.join([word for word in x.split() if word not in (stop_words)]))

Test_Df['Msg_without_Stopwords']=Test_Df['Msg_with_Stopwords'].apply(lambda x: ' '.join([word for word in x.split() if word not in (stop_words)]))

In [28]:
Train_Df['Msg_with_Stopwords_Len'] = [len(x.split()) for x in Train_Df['Msg_with_Stopwords'].tolist()]
Train_Df['Msg_without_Stopwords_Len'] = [len(x.split()) for x in Train_Df['Msg_without_Stopwords'].tolist()]

Dev_Df['Msg_with_Stopwords_Len'] = [len(x.split()) for x in Dev_Df['Msg_with_Stopwords'].tolist()]
Dev_Df['Msg_without_Stopwords_Len'] = [len(x.split()) for x in Dev_Df['Msg_without_Stopwords'].tolist()]

Test_Df['Msg_with_Stopwords_Len'] = [len(x.split()) for x in Test_Df['Msg_with_Stopwords'].tolist()]
Test_Df['Msg_without_Stopwords_Len'] = [len(x.split()) for x in Test_Df['Msg_without_Stopwords'].tolist()]

In [29]:
Train_Data=pd.concat([Mal_train, Train_Df], axis=1)

Dev_Data=pd.concat([Mal_Dev, Dev_Df], axis=1)

Test_Data=pd.concat([Mal_test, Test_Df], axis=1)

In [30]:
Train_Data.shape, Dev_Data.shape, Test_Data.shape

((35656, 6), (3962, 6), (4402, 6))

In [31]:
Test_Data

id                                               text  \
0        Tam_1  வீர செங்குந்தர் சார்பாக இந்த திரைப்படம் வெற்றி...   
1        Tam_2  Teruk ah irukku .... mokke movie .. waste of time   
2        Tam_3  manitha samuthaayam amaipil irunthu intha pada...   
3        Tam_4                                   JJ mam we miss u   
4        Tam_5          Subtitle me traller dekhne wale like karo   
...        ...                                                ...   
4397  Tam_4398  Ithukum dislike potta kammanattti koovaingalam...   
4398  Tam_4399  Suyama Sinthikiravan than super Hero Seama dia...   
4399  Tam_4400           Super thalaiva.... Nee mass dha eppavume   
4400  Tam_4401  பெண்ணை அடிமையாக்க நினைக்கும் இந்த படம் தோல்வித...   
4401  Tam_4402  Semma thalaiva alu athikama akirukum enimale e...   

                                     Msg_with_Stopwords  \
0     வீர செங்குந்தர் சார்பாக இந்த திரைப்படம் வெற்றி...   
1             Teruk ah irukku mokke movie waste of time   
2     manitha samuthaayam amaipil irunthu intha pada...   
3                                      JJ mam we miss u   
4             Subtitle me traller dekhne wale like karo   
...                                                 ...   
4397  Ithukum dislike potta kammanattti koovaingalam...   
4398  Suyama Sinthikiravan than super Hero Seama dia...   
4399               Super thalaiva Nee mass dha eppavume   
4400  பெண்ணை அடிமையாக்க நினைக்கும் இந்த படம் தோல்வித...   
4401  Semma thalaiva alu athikama akirukum enimale e...   

                                  Msg_without_Stopwords  \
0     வீர செங்குந்தர் சார்பாக இந்த திரைப்படம் வெற்றி...   
1                Teruk ah irukku mokke movie waste time   
2     manitha samuthaayam amaipil irunthu intha pada...   
3                                         JJ mam miss u   
4                Subtitle traller dekhne wale like karo   
...                                                 ...   
4397  Ithukum dislike potta kammanattti koovaingalam...   
4398     Suyama Sinthikiravan super Hero Seama dialogue   
4399               Super thalaiva Nee mass dha eppavume   
4400  பெண்ணை அடிமையாக்க நினைக்கும் இந்த படம் தோல்வித...   
4401  Semma thalaiva alu athikama akirukum enimale e...   

      Msg_with_Stopwords_Len  Msg_without_Stopwords_Len  
0                         15                         15  
1                          8                          7  
2                          9                          9  
3                          5                          4  
4                          7                          6  
...                      ...                        ...  
4397                      10                         10  
4398                       7                          6  
4399                       6                          6  
4400                       7                          7  
4401                      11                          9  

[4402 rows x 6 columns]

In [32]:
Train_Data=Train_Data.drop(['Msg_with_Stopwords','Msg_without_Stopwords_Len', 'Msg_with_Stopwords_Len', 'text'], axis=1)

Dev_Data=Dev_Data.drop(['Msg_with_Stopwords','Msg_without_Stopwords_Len', 'Msg_with_Stopwords_Len', 'text'], axis=1)

Test_Data=Test_Data.drop(['Msg_with_Stopwords','Msg_without_Stopwords_Len', 'Msg_with_Stopwords_Len', 'text'], axis=1)


In [33]:
# Dataset Train & Dev are : (1) Message_Without_Stopwords and (2) labels
# Only Test is : (1) Message_Without_Stopwords
Train_Data.shape, Dev_Data.shape, Test_Data.shape

((35656, 2), (3962, 2), (4402, 2))

In [34]:
Train_Data[78:90]

category                              Msg_without_Stopwords
78   Positive  Cmnt section pona like like nu uyira vanguvanu...
79  not-Tamil                      tranding kyu ni aa rahi h yrr
80   Positive                Mrana Mass iliya suppr star DHANUSH
81   Positive        vera level pakka blockbuster sure best anna
82   Positive  வன்னியர் சாதி சார்பாக படம் வெற்றி பெற வாழ்த்து...
83   Positive             Yuvan bgm vera level love u yuvan anna
84   Positive  Rishi Richard Fans Club சார்பாக படம் வெற்றியடை...
85   Positive  ஜெயம் ரவி அண்ணா வாழ்த்துக்கள் நீங்க இதே மாதிரி...
86   Positive  படம் வெற்றிகரமாக அமைய வாழ்த்துக்கள் செளராஷ்டிர...
87   Negative                                 rubbish Ayoooo ayo
88   Positive          தரமா டீஸர் அதேபோல் படமும் தரமாக இருக்கும்
89   Positive  Maximum intha mathiri msg kudukara reality Fil...

In [35]:
Dev_Data

category                              Msg_without_Stopwords
0      unknown_state          songsuperbkamallllla I'm waiting dis song
1           Positive  கணத்ததோர் அகமுடையார் சார்பாக படம் வெற்றி அடைய ...
2           Positive  Thalavia neenga veera level boss neega thaan m...
3           Positive               Oru padam patha fellvera level music
4      unknown_state  Hairstyle mattama iruku Adhu mattum mathirukal...
...              ...                                                ...
3957        Positive  நாடக காதல் மிகப்பெரிய தவறு இதை தான் இந்த படம் ...
3958       not-Tamil                                 So long see suriya
3959  Mixed_feelings  Super sK ettan adipoli makkalle dec waiting sK...
3960        Positive              Minimum viewa vara veppoma thala fans
3961        Positive                         Vera Vera level da version

[3962 rows x 2 columns]

In [36]:
Test_Data

id                              Msg_without_Stopwords
0        Tam_1  வீர செங்குந்தர் சார்பாக இந்த திரைப்படம் வெற்றி...
1        Tam_2             Teruk ah irukku mokke movie waste time
2        Tam_3  manitha samuthaayam amaipil irunthu intha pada...
3        Tam_4                                      JJ mam miss u
4        Tam_5             Subtitle traller dekhne wale like karo
...        ...                                                ...
4397  Tam_4398  Ithukum dislike potta kammanattti koovaingalam...
4398  Tam_4399     Suyama Sinthikiravan super Hero Seama dialogue
4399  Tam_4400               Super thalaiva Nee mass dha eppavume
4400  Tam_4401  பெண்ணை அடிமையாக்க நினைக்கும் இந்த படம் தோல்வித...
4401  Tam_4402  Semma thalaiva alu athikama akirukum enimale e...

[4402 rows x 2 columns]

In [37]:
from sklearn import preprocessing
le = preprocessing.LabelEncoder()

Train_Data['Nlabel']=le.fit_transform(Train_Data['category']) 
Dev_Data['Nlabel']=le.fit_transform(Dev_Data['category']) 

In [38]:
Train_Data['Nlabel'].value_counts()

2    20070
4     5628
1     4271
0     4020
3     1667
Name: Nlabel, dtype: int64

In [39]:
2    20069
5     5628
1     4271
0     4020
4     1667
3        1

Positive      2    20070
unknown_state  4    5628
Negative      1     4271
Mixed_feelings  0   4020
not-Tamil     3     1667

Positive          706 2
unknown_state     580 4
Negative          237 1
not-malayalam     141 3
Mixed_feelings    102 0

SyntaxError: invalid syntax (<ipython-input-39-14ccf4c00261>, line 1)

In [39]:
Dev_Data['Nlabel'].value_counts()

2    2257
4     611
1     480
0     438
3     176
Name: Nlabel, dtype: int64

In [40]:
Train_Data

category                              Msg_without_Stopwords  \
0       unknown_state  Vani bhojam fans hit like solli like Vangida V...   
1            Positive                                I love ajith I like   
2            Positive  ennaya trailer Ku mudi Ellam nikkudhu Vera lev...   
3            Positive            Vijay Annaa Ur Maassssss Therrrrriiiiii   
4            Negative                        நம்ப நடே நாசாமா தான் போச்சி   
...               ...                                                ...   
35651  Mixed_feelings  ஒருவர் செய்த தவறுக்காக எல்லாரையும் பழி போடுவது...   
35652        Positive  Repeated Mode Watching Theri Trailer Theri son...   
35653        Negative  yeevanikuachu pula kutti erudhal oodi poyidu m...   
35654        Positive                    She looking like laughing budha   
35655        Positive                     NORTH OR SOUTH KGF IS THE BEST   

       Nlabel  
0           4  
1           2  
2           2  
3           2  
4           1  
...       ...  
35651       0  
35652       2  
35653       1  
35654       2  
35655       2  

[35656 rows x 3 columns]

In [41]:
Dev_Data[50:70]

category                              Msg_without_Stopwords  Nlabel
50   unknown_state  Bigillike Kaithi cmnt Idhu epdi irruku vj fns ...       4
51   unknown_state  Camne bole trending kt malaysia Yg dri malaysi...       4
52        Negative             Auto like alludhu vekkama illayada Thu       1
53        Positive  Itha paartha kaapan maathiri illa adhavan maat...       2
54  Mixed_feelings                          k likes thala Ajith storm       0
55        Positive  மோகன் அண்ணா மேன்மேலும் திரைப்படத்துறையில் வளர ...       2
56        Positive  Namba kita kaadu iruntha eduthukuvango ruuva i...       2
57        Negative  சாதிய ஒழிக்கனுனு koovitu சாதி வெறி புடிச்ச தேவ...       1
58        Positive  I think sema line Ravi pidisu irukaaru Sema jo...       2
59   unknown_state  Like like nu solravanga enaku like potaala k l...       4
60  Mixed_feelings  அடுத்தவங்க இசையை பயன்படுத்துகிறவங்க எல்லாம் ஆண...       0
61        Positive  My fevret Hero Akshay kumar Supar Dupar hogi M...       2
62        Positive  VANNIYAR kottai dharmapuri padaiyachi Vanniyar...       2
63        Positive                   Pwoli rakshit shetty kerala fans       2
64   unknown_state  Simbu zindhabad kannada power puneeth rajkumar...       4
65        Negative  Avalavu nalla illai I'm superstar hater kooda ...       1
66   unknown_state                          Str da goyala str fans da       4
67  Mixed_feelings             Padoo mass ah iruka podhu tn la odadhu       0
68   unknown_state  Fans show clock porom massss kaaatrom surya an...       4
69        Positive                      Waiting seeeeeeeeee Anna mass       2

In [42]:
Test_Data

id                              Msg_without_Stopwords
0        Tam_1  வீர செங்குந்தர் சார்பாக இந்த திரைப்படம் வெற்றி...
1        Tam_2             Teruk ah irukku mokke movie waste time
2        Tam_3  manitha samuthaayam amaipil irunthu intha pada...
3        Tam_4                                      JJ mam miss u
4        Tam_5             Subtitle traller dekhne wale like karo
...        ...                                                ...
4397  Tam_4398  Ithukum dislike potta kammanattti koovaingalam...
4398  Tam_4399     Suyama Sinthikiravan super Hero Seama dialogue
4399  Tam_4400               Super thalaiva Nee mass dha eppavume
4400  Tam_4401  பெண்ணை அடிமையாக்க நினைக்கும் இந்த படம் தோல்வித...
4401  Tam_4402  Semma thalaiva alu athikama akirukum enimale e...

[4402 rows x 2 columns]

In [43]:
X_train=Train_Data['Msg_without_Stopwords']
y_train=Train_Data['Nlabel']

X_Dev=Dev_Data['Msg_without_Stopwords']
y_Dev=Dev_Data['Nlabel']

X_test=Test_Data['Msg_without_Stopwords']

In [44]:
X_train.shape, y_train.shape, X_Dev.shape, y_Dev.shape, X_test.shape

((35656,), (35656,), (3962,), (3962,), (4402,))

Not_offensive                           14153     0
not-malayalam                            1287     4
Offensive_Targeted_Insult_Individual      239     2
Offensive_Untargetede                     191     3
Offensive_Targeted_Insult_Group           140     1


Not_offensive                           1779      0
not-malayalam                            163      4
Offensive_Targeted_Insult_Individual      24      2
Offensive_Untargetede                     20      3
Offensive_Targeted_Insult_Group           13      1

In [141]:
from sklearn.feature_extraction.text import TfidfVectorizer
tfidf=TfidfVectorizer(analyzer='word', lowercase=True , ngram_range=(5,6), max_features=10000)
X=tfidf.fit_transform(X_train) 
Y=y_train
tfidf

TfidfVectorizer(max_features=10000, ngram_range=(5, 6))

In [142]:
char_tfidf=TfidfVectorizer(analyzer='char', lowercase=True, ngram_range=(5,6), max_features=10000)
X_char=char_tfidf.fit_transform(X_train) 
char_tfidf

TfidfVectorizer(analyzer='char', max_features=10000, ngram_range=(5, 6))

In [143]:
len(tfidf.vocabulary_), len(char_tfidf.vocabulary_), len(tfidf.get_feature_names()), len(char_tfidf.get_feature_names())


(10000, 10000, 10000, 10000)

In [144]:
tfidf.vocabulary_

{'vera level da deii getrajinified': 4806,
 'vijay hair hair setup paaa': 7123,
 'vettri pera vaazhthukkaljaadhi paarkadha manidhargal': 5510,
 'vettri pera vaazhthukkaljaadhi paarkadha manidhargal samoogathil': 5511,
 'மந தப ரம வட டம': 9439,
 'தவ சம யம படம அட': 9057,
 'சம யம படம அட கள': 8746,
 'தவ சம யம படம அட கள': 9058,
 'vazthukalnamakkal kongu vellar sangam sarbil': 2833,
 'vazthukalnamakkal kongu vellar sangam sarbil vazthukal': 2834,
 'அற மக கள கழ கள': 7753,
 'மக கள கழ கள கள': 9385,
 'கள கழ கள கள நன': 8547,
 'அற மக கள கழ கள கள': 7754,
 'மக கள கழ கள கள நன': 9386,
 'venalum pakalam storyla neriya vishayam': 3961,
 'venalum pakalam storyla neriya vishayam iruku': 3962,
 'neraya neraya neraya neraya neraya': 1095,
 'neraya neraya neraya neraya powerful': 1098,
 'neraya neraya neraya neraya neraya neraya': 1096,
 'neraya neraya neraya neraya neraya powerful': 1097,
 'vera level dhanush anna gv': 4841,
 'vera level dhanush anna gv bgm': 4842,
 'vera level love yuvan anna': 5208,
 'wow 

In [145]:
char_tfidf.vocabulary_

{' fans': 234,
 'fans ': 3566,
 'ans h': 2582,
 'ns hi': 5609,
 's hit': 6345,
 ' hit ': 282,
 'hit l': 3885,
 'it li': 4294,
 't lik': 6557,
 ' like': 496,
 'like ': 4874,
 'ike s': 4100,
 'e sol': 3262,
 ' soll': 891,
 'solli': 6478,
 'olli ': 5721,
 'i lik': 3989,
 'ike v': 4103,
 'ke va': 4523,
 'e van': 3278,
 ' vang': 1018,
 'da ve': 3071,
 'a ven': 1911,
 ' vend': 1051,
 'vendi': 7386,
 'ndiya': 5421,
 'diyat': 3187,
 'iyath': 4382,
 'yathu': 7549,
 ' fans ': 235,
 'fans h': 3568,
 'ans hi': 2583,
 'ns hit': 5610,
 's hit ': 6346,
 ' hit l': 284,
 'hit li': 3886,
 'it lik': 4295,
 't like': 6558,
 ' like ': 497,
 'like s': 4882,
 ' solli': 893,
 'solli ': 6479,
 'i like': 3990,
 'like v': 4884,
 'ike va': 4104,
 'a vend': 1912,
 ' vendi': 1052,
 'diyath': 3188,
 'iyathu': 4384,
 'i lov': 3991,
 ' love': 503,
 'love ': 4949,
 ' ajit': 34,
 'ajith': 2176,
 'jith ': 4408,
 'i love': 3992,
 ' love ': 504,
 ' ajith': 35,
 'ajith ': 2177,
 'a tra': 1878,
 ' trai': 970,
 'trail': 6815,

In [146]:
X, X_char

(<35656x10000 sparse matrix of type '<class 'numpy.float64'>'
 	with 17204 stored elements in Compressed Sparse Row format>,
 <35656x10000 sparse matrix of type '<class 'numpy.float64'>'
 	with 1801411 stored elements in Compressed Sparse Row format>)

In [147]:
from scipy.sparse import hstack
train=hstack([X,X_char])
train

<35656x20000 sparse matrix of type '<class 'numpy.float64'>'
	with 1818615 stored elements in COOrdinate format>

In [148]:
Xt=tfidf.transform(X_Dev) 
Xt_char=char_tfidf.transform(X_Dev)

In [149]:
Xt, Xt_char

(<3962x10000 sparse matrix of type '<class 'numpy.float64'>'
 	with 515 stored elements in Compressed Sparse Row format>,
 <3962x10000 sparse matrix of type '<class 'numpy.float64'>'
 	with 197286 stored elements in Compressed Sparse Row format>)

In [150]:
Dev=hstack([Xt,Xt_char])
Dev

<3962x20000 sparse matrix of type '<class 'numpy.float64'>'
	with 197801 stored elements in COOrdinate format>

In [151]:
test_Mal_Hasoc_word=tfidf.transform(X_test)
test_Mal_Hasoc_char=char_tfidf.transform(X_test)
test_Mal_Hasoc_word_char=hstack([test_Mal_Hasoc_word,test_Mal_Hasoc_char])

In [152]:
from sklearn.metrics import confusion_matrix
from sklearn.metrics import accuracy_score
from sklearn.metrics import classification_report
import seaborn as sn
import matplotlib.pyplot as plt
from sklearn.model_selection import GridSearchCV
from sklearn.metrics import precision_score, recall_score, f1_score

In [153]:
def trainer(model_obj, param, X, X_char, train, Xt, Xt_char, Dev, Y, y_Dev):
   
    model_obj.fit(X,Y)
#Training Classifier with TF-IDF Features Word n-gram
    print("Accuracy of Word n-gram(1,2) :", (model_obj.score(Xt, y_Dev)))
#Printing Classification Report
    predict=model_obj.predict(Xt)
    print(classification_report(y_Dev, predict))
#Printing Confusion Matrix
    conf_mat=confusion_matrix(y_Dev, predict)
    print("Confusion Matrix")
    print(conf_mat)
    print(" ")
    word=model_obj.predict(test_Mal_Hasoc_word)
    
    pre=precision_score(y_Dev, predict, average='micro')
    rec=recall_score(y_Dev, predict, average='micro')
    f1s=f1_score(y_Dev, predict, average='micro')
    
    print("Precision : ", pre)
    print("Recall : ", rec)
    print("F1 Score : ", f1s)

#Training Classifier with TF-IDF Features Char n-gram
    model_obj.fit(X_char, Y)
    print("Accuracy of Char n-gram(1,5) :", (model_obj.score(Xt_char, y_Dev)))
#Printing Classification Report
    predict=model_obj.predict(Xt_char)
    print(classification_report(y_Dev, predict))
#Printing Confusion Matrix
    conf_mat=confusion_matrix(y_Dev, predict)
    print("Confusion Matrix")
    print(conf_mat)
    print(" ")
    char=model_obj.predict(test_Mal_Hasoc_char)
    
    pre=precision_score(y_Dev, predict, average='micro')
    rec=recall_score(y_Dev, predict, average='micro')
    f1s=f1_score(y_Dev, predict, average='micro')
    
    print("Precision : ", pre)
    print("Recall : ", rec)
    print("F1 Score : ", f1s)
        
              
#Training Classifier with TF-IDF Features Combine Word n-gram & Char n-gram
    model_obj.fit(train, Y)
    print("Accuracy of Combinining word & Char n-gram", (model_obj.score(Dev, y_Dev)))
#Printing Classification Report
    predict=model_obj.predict(Dev)
    print(classification_report(y_Dev, predict))
#Printing Confusion Matrix
    conf_mat=confusion_matrix(y_Dev, predict)
    print("Confusion Matrix")
    print(conf_mat)
    print(" ")
    word_char=model_obj.predict(test_Mal_Hasoc_word_char)
    
    pre=precision_score(y_Dev, predict,  average='micro')
    rec=recall_score(y_Dev, predict, average='micro')
    f1s=f1_score(y_Dev, predict, average='micro')
    
    print("Precision : ", pre,)
    print("Recall : ", rec)
    print("F1 Score : ", f1s)

    
    return word, char, word_char

In [154]:
from sklearn.naive_bayes import MultinomialNB
MNB=MultinomialNB(alpha=2.0)
param={'alpha':[0.5,0.6,0.7,0.8,0.9,1.0,1.1,1.2,2.0]}
print("Multi Nominal Bayes")
trainer(MNB,param, X, X_char, train , Xt, Xt_char, Dev, Y, y_Dev)
#w_mnb,c_mnb,wc_mnb=trainer(MNB,param, X, X_char, train , Xt, Xt_char, Dev, Y, y_Dev)
# w_mnb,c_mnb,wc_mnb=trainer(MNB,param, X, X_char, train, traintest, Y, y_test)

Multi Nominal Bayes
Accuracy of Word n-gram(1,2) : 0.5704189803129732
              precision    recall  f1-score   support

           0       0.50      0.00      0.00       438
           1       0.00      0.00      0.00       480
           2       0.57      1.00      0.73      2257
           3       0.00      0.00      0.00       176
           4       0.67      0.01      0.01       611

    accuracy                           0.57      3962
   macro avg       0.35      0.20      0.15      3962
weighted avg       0.48      0.57      0.42      3962

Confusion Matrix
[[   1    0  436    0    1]
 [   0    0  480    0    0]
 [   1    0 2255    0    1]
 [   0    0  176    0    0]
 [   0    0  607    0    4]]
 
Precision :  0.5704189803129732
Recall :  0.5704189803129732
F1 Score :  0.5704189803129732
Accuracy of Char n-gram(1,5) : 0.57218576476527
              precision    recall  f1-score   support

           0       0.24      0.13      0.17       438
           1       0.33      0.4

e:\Anaconda3\lib\site-packages\sklearn\metrics\_classification.py:1221: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


Accuracy of Combinining word & Char n-gram 0.6060070671378092
              precision    recall  f1-score   support

           0       0.33      0.10      0.15       438
           1       0.43      0.28      0.34       480
           2       0.66      0.88      0.75      2257
           3       0.67      0.21      0.32       176
           4       0.44      0.34      0.39       611

    accuracy                           0.61      3962
   macro avg       0.51      0.36      0.39      3962
weighted avg       0.56      0.61      0.56      3962

Confusion Matrix
[[  42   53  286    6   51]
 [  15  135  282    3   45]
 [  43   78 1978    7  151]
 [   0    2  122   37   15]
 [  29   45  326    2  209]]
 
Precision :  0.6060070671378092
Recall :  0.6060070671378092
F1 Score :  0.6060070671378092


(array([2, 2, 2, ..., 2, 2, 2]),
 array([2, 2, 2, ..., 2, 0, 2]),
 array([2, 2, 2, ..., 2, 2, 2]))

In [155]:
#from sklearn.tree import DecisionTreeClassifier

#DTC=DecisionTreeClassifier()
#trainer(DTC,param, X, X_char, train , Xt, Xt_char, Dev, Y, y_Dev)

In [156]:
#from sklearn.utils import class_weight
#import numpy as np
#class_weights = class_weight.compute_class_weight('balanced',np.unique(Y),Y)

In [157]:
#class_weights

In [158]:
#class_weights = class_weight.compute_class_weight('balanced',np.unique(Y),Y)

In [159]:
#from sklearn.naive_bayes import MultinomialNB
#NB11=MultinomialNB()


In [160]:
#class_weights = class_weight.compute_class_weight('balanced',np.unique(Y),Y)

In [161]:
from imblearn.over_sampling import SMOTE

In [162]:
smote = SMOTE()

In [163]:
x_smote_Word, y_smote_Word = smote.fit_resample(X, Y)
x_smote_Char, y_smote_Char = smote.fit_resample(X_char, Y)
x_smote_Comb, y_smote_Comb = smote.fit_resample(train, Y)

In [164]:
Y.value_counts()

2    20070
4     5628
1     4271
0     4020
3     1667
Name: Nlabel, dtype: int64

In [165]:
x_smote_Word.shape, y_smote_Word.shape, x_smote_Char.shape, y_smote_Char.shape, x_smote_Comb.shape, y_smote_Comb.shape

((100350, 10000),
 (100350,),
 (100350, 10000),
 (100350,),
 (100350, 20000),
 (100350,))

# Complete Feature set
((100350, 339558),
 (100350,),
 (100350, 724580),
 (100350,),
 (100350, 1064138),
 (100350,))

In [166]:
# Development Dataset
x_smote_Word_Dev, y_smote_Word_Dev = smote.fit_resample(Xt, y_Dev)
x_smote_Char_Dev, y_smote_Char_Dev = smote.fit_resample(Xt_char, y_Dev)
x_smote_Comb_Dev, y_smote_Comb_Dev = smote.fit_resample(Dev, y_Dev)

In [167]:
x_smote_Word_Dev.shape, y_smote_Word_Dev.shape, x_smote_Char_Dev.shape, y_smote_Char_Dev.shape, x_smote_Comb_Dev.shape, y_smote_Comb_Dev.shape

((11285, 10000), (11285,), (11285, 10000), (11285,), (11285, 20000), (11285,))

# complete feature set
((11285, 339558),
 (11285,),
 (11285, 724580),
 (11285,),
 (11285, 1064138),
 (11285,))

In [168]:
from sklearn.model_selection import GridSearchCV


In [169]:
def trainer1111(model_obj, param, x_smote_Word,y_smote_Word, x_smote_Char,y_smote_Char, x_smote_Comb ,y_smote_Comb, x_smote_Word_Dev, y_smote_Word_Dev, x_smote_Char_Dev,y_smote_Char_Dev, x_smote_Comb_Dev, y_smote_Comb_Dev):
    # Training Classifier  with TF-IDF Features Word n-gram(1,2)
    GSCV=GridSearchCV(model_obj, param , cv=10,return_train_score=False)
    GSCV.fit(x_smote_Word, y_smote_Word)
    print("Accuracy of Word n-gram(1,1) :", (GSCV.score(x_smote_Word_Dev, y_smote_Word_Dev)))
    #Printing Classification Report
    predict=GSCV.predict(x_smote_Word_Dev)
    print(classification_report(y_smote_Word_Dev, predict))
    #Printing Confusion Matrix
    conf_mat=confusion_matrix(y_smote_Word_Dev, predict)
    print("Best Parameters ", GSCV.best_params_)
    print("Best Score ", GSCV.best_score_)
    print(" ")
    print("Confusion Matrix")
    print(conf_mat)
    print(" ")
    word=GSCV.predict(test_Mal_Hasoc_word)
 
    accuracy = accuracy_score(y_smote_Word_Dev, predict)
    recall = recall_score(y_smote_Word_Dev, predict, average='micro')
    precision = precision_score(y_smote_Word_Dev, predict, average='micro')
    f1score = f1_score(y_smote_Word_Dev, predict, average='micro')
    print ('accuracy: {}, recall: {}, precision: {}, f1-score: {}'.format(accuracy, recall, precision, f1score))
    
    
    #Training Classifier with TF-IDF Features Char n-gram(1,5)
    GSCV.fit(x_smote_Char, y_smote_Char)
    print("Accuracy of Char n-gram(1,5) :", (GSCV.score(x_smote_Char_Dev, y_smote_Char_Dev)))
    predict=GSCV.predict(x_smote_Char_Dev)
    print(classification_report(y_smote_Char_Dev, predict))
   #Printing Confusion Matrix
    conf_mat=confusion_matrix(y_smote_Char_Dev, predict)
    print("Best Parameters ", GSCV.best_params_)
    print("Best Score ", GSCV.best_score_)
    print(" ")
    print("Confusion Matrix")
    print(conf_mat)
    print(" ")
    char=GSCV.predict(test_Mal_Hasoc_char)
    
    accuracy = accuracy_score(y_smote_Char_Dev, predict)
    recall = recall_score(y_smote_Char_Dev, predict, average='micro')
    precision = precision_score(y_smote_Char_Dev, predict, average='micro')
    f1score = f1_score(y_smote_Char_Dev, predict, average='micro')
    print ('accuracy: {}, recall: {}, precision: {}, f1-score: {}'.format(accuracy, recall, precision, f1score))
    
    
   
    
    
    #Training Classifier with TF-IDF Features Combine Word n-gram & Char n-gram
    GSCV.fit(x_smote_Comb, y_smote_Comb)
    print("Accuracy of Combinining word & Char n-gram", (GSCV.score(x_smote_Comb, y_smote_Comb)))
    #Printing Classification Report
    predict=GSCV.predict(x_smote_Comb_Dev)
    print(classification_report(y_smote_Comb_Dev, predict))
    #Printing Confusion Matrix
    conf_mat=confusion_matrix(y_smote_Comb_Dev, predict)
    print("Best Parameters ", GSCV.best_params_)
    print("Best Score ", GSCV.best_score_)
    print(" ")
    print("Confusion Matrix")
    print(conf_mat)
    print(" ")
    word_char=GSCV.predict(test_Mal_Hasoc_word_char)
    
    accuracy = accuracy_score(y_smote_Comb_Dev, predict)
    recall = recall_score(y_smote_Comb_Dev, predict, average='micro')
    precision = precision_score(y_smote_Comb_Dev, predict, average='micro')
    f1score = f1_score(y_smote_Comb_Dev, predict, average='micro')
    print ('accuracy: {}, recall: {}, precision: {}, f1-score: {}'.format(accuracy, recall, precision, f1score))
    
    return word, char, word_char

In [170]:
from sklearn.naive_bayes import MultinomialNB
MNB11=MultinomialNB()
param={'alpha':[0.5,0.6,0.7,0.8,0.9,1.0,1.1,1.2,2.0]}
print("Multi Nominal Bayes")
#trainer1111(MNB11,param, x_smote_Word,y_smote_Word, x_smote_Char,y_smote_Char, x_smote_Comb ,y_smote_Comb, x_smote_Word_Dev, y_smote_Word_Dev, x_smote_Char_Dev,y_smote_Char_Dev, x_smote_Comb_Dev, y_smote_Comb_Dev)
w_mnb,c_mnb,wc_mnb=trainer1111(MNB11,param, x_smote_Word,y_smote_Word, x_smote_Char,y_smote_Char, x_smote_Comb ,y_smote_Comb, x_smote_Word_Dev, y_smote_Word_Dev, x_smote_Char_Dev,y_smote_Char_Dev, x_smote_Comb_Dev, y_smote_Comb_Dev)

Multi Nominal Bayes
Accuracy of Word n-gram(1,1) : 0.21657066902968541
              precision    recall  f1-score   support

           0       0.20      0.98      0.34      2257
           1       0.25      0.00      0.00      2257
           2       0.57      0.04      0.07      2257
           3       0.86      0.02      0.05      2257
           4       0.74      0.03      0.07      2257

    accuracy                           0.22     11285
   macro avg       0.52      0.22      0.10     11285
weighted avg       0.52      0.22      0.10     11285

Best Parameters  {'alpha': 0.5}
Best Score  0.27765819631290484
 
Confusion Matrix
[[2222    0   26    0    9]
 [2253    4    0    0    0]
 [2144    3   86    5   19]
 [2187    0   16   54    0]
 [2144    9   22    4   78]]
 
accuracy: 0.21657066902968541, recall: 0.21657066902968541, precision: 0.21657066902968541, f1-score: 0.2165706690296854
Accuracy of Char n-gram(1,5) : 0.49073992024811697
              precision    recall  f1-scor

# Multi Nominal Bayes
Accuracy of Word n-gram(1,1) : 0.22374833850243686
              precision    recall  f1-score   support

           0       0.20      0.95      0.33      2257
           1       0.47      0.01      0.02      2257
           2       0.45      0.06      0.11      2257
           3       0.96      0.05      0.09      2257
           4       0.58      0.05      0.08      2257

    accuracy                           0.22     11285
   macro avg       0.53      0.22      0.13     11285
weighted avg       0.53      0.22      0.13     11285

Best Parameters  {'alpha': 0.6}
Best Score  0.7394419531639264
 
Confusion Matrix
[[2143    7   76    0   31]
 [2228   23    6    0    0]
 [2076    5  145    5   26]
 [2102    0   25  111   19]
 [2069   14   71    0  103]]
 
accuracy: 0.22374833850243686, recall: 0.22374833850243686, precision: 0.22374833850243686, f1-score: 0.22374833850243686
Accuracy of Char n-gram(1,5) : 0.5654408506867523
              precision    recall  f1-score   support

           0       0.40      0.38      0.39      2257
           1       0.51      0.62      0.56      2257
           2       0.51      0.59      0.54      2257
           3       0.84      0.80      0.82      2257
           4       0.61      0.44      0.51      2257

    accuracy                           0.57     11285
   macro avg       0.57      0.57      0.56     11285
weighted avg       0.57      0.57      0.56     11285

Best Parameters  {'alpha': 0.5}
Best Score  0.7906228201295467
 
Confusion Matrix
[[ 847  646  511   76  177]
 [ 495 1410  188   47  117]
 [ 328  301 1321  107  200]
 [   9   53  249 1811  135]
 [ 438  368  345  114  992]]
 
accuracy: 0.5654408506867523, recall: 0.5654408506867523, precision: 0.5654408506867523, f1-score: 0.5654408506867523
Accuracy of Combinining word & Char n-gram 0.9504434479322371
              precision    recall  f1-score   support

           0       0.42      0.30      0.35      2257
           1       0.53      0.58      0.55      2257
           2       0.45      0.68      0.54      2257
           3       0.85      0.75      0.80      2257
           4       0.55      0.44      0.49      2257

    accuracy                           0.55     11285
   macro avg       0.56      0.55      0.55     11285
weighted avg       0.56      0.55      0.55     11285

Best Parameters  {'alpha': 0.5}
Best Score  0.8880318883906329
 
Confusion Matrix
[[ 676  564  710   51  256]
 [ 355 1305  348   42  207]
 [ 195  228 1529   96  209]
 [  19   52  335 1695  156]
 [ 376  327  452  106  996]]
 
accuracy: 0.549490474080638, recall: 0.549490474080638, precision: 0.549490474080638, f1-score: 0.549490474080638



In [175]:
c_mnb

array([2, 1, 2, ..., 2, 0, 2])

In [ ]:
Positive      2    20070
unknown_state  4    5628
Negative      1     4271
Mixed_feelings  0   4020
not-Tamil     3     1667

In [177]:
c_mnb=pd.DataFrame(c_mnb, columns=["Word_MNB"])
FINAL=pd.concat([X_test, c_mnb], axis=1)
FINAL=FINAL.rename(columns={"Msg_without_Stopwords": "text", "Word_MNB": "label"})

FINAL['label'] = FINAL['label'].map({0 : 'Mixed_feelings', 1 : 'Negative', 2: 'Positive', 3 :'not-Tamil', 4 : 'unknown_state'})
FINAL

text           label
0     வீர செங்குந்தர் சார்பாக இந்த திரைப்படம் வெற்றி...        Positive
1                Teruk ah irukku mokke movie waste time        Negative
2     manitha samuthaayam amaipil irunthu intha pada...        Positive
3                                         JJ mam miss u        Positive
4                Subtitle traller dekhne wale like karo       not-Tamil
...                                                 ...             ...
4397  Ithukum dislike potta kammanattti koovaingalam...        Negative
4398     Suyama Sinthikiravan super Hero Seama dialogue        Positive
4399               Super thalaiva Nee mass dha eppavume        Positive
4400  பெண்ணை அடிமையாக்க நினைக்கும் இந்த படம் தோல்வித...  Mixed_feelings
4401  Semma thalaiva alu athikama akirukum enimale e...        Positive

[4402 rows x 2 columns]

# Positive          6421 2
unknown_state     5279 4
Negative          2105 1
not-malayalam     1157 3
Mixed_feelings     926 0

In [178]:
FINAL.to_csv("MNB_Tamil_HASOC21.tsv")

In [172]:
from sklearn.linear_model import LogisticRegression
LR=LogisticRegression()
param={'solver':('newton-cg', 'lbfgs', 'liblinear', 'sag', 'saga'), 'multi_class':('ovr', 'auto')}
print("Multi Nominal Bayes")
#trainer1111(MNB11,param, x_smote_Word,y_smote_Word, x_smote_Char,y_smote_Char, x_smote_Comb ,y_smote_Comb, x_smote_Word_Dev, y_smote_Word_Dev, x_smote_Char_Dev,y_smote_Char_Dev, x_smote_Comb_Dev, y_smote_Comb_Dev)
w_mnb,c_mnb,wc_mnb=trainer1111(LR,param, x_smote_Word,y_smote_Word, x_smote_Char,y_smote_Char, x_smote_Comb ,y_smote_Comb, x_smote_Word_Dev, y_smote_Word_Dev, x_smote_Char_Dev,y_smote_Char_Dev, x_smote_Comb_Dev, y_smote_Comb_Dev)

Multi Nominal Bayes


e:\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:764: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)
e:\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:764: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_

Accuracy of Word n-gram(1,1) : 0.22233052724856003
              precision    recall  f1-score   support

           0       0.56      0.01      0.02      2257
           1       0.21      1.00      0.34      2257
           2       0.61      0.04      0.07      2257
           3       0.85      0.02      0.04      2257
           4       0.77      0.04      0.07      2257

    accuracy                           0.22     11285
   macro avg       0.60      0.22      0.11     11285
weighted avg       0.60      0.22      0.11     11285

Best Parameters  {'multi_class': 'auto', 'solver': 'newton-cg'}
Best Score  0.284853014449427
 
Confusion Matrix
[[  24 2198   26    0    9]
 [   0 2257    0    0    0]
 [   9 2138   88    5   17]
 [   0 2189   16   52    0]
 [  10 2140   15    4   88]]
 
accuracy: 0.22233052724856003, recall: 0.22233052724856003, precision: 0.22233052724856003, f1-score: 0.22233052724856003


e:\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:764: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)
e:\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:764: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_

e:\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:764: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)
e:\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:764: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_

e:\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:764: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)
e:\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:764: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_

e:\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:764: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)
e:\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:764: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_

Accuracy of Char n-gram(1,5) : 0.5208684093929996
              precision    recall  f1-score   support

           0       0.36      0.32      0.34      2257
           1       0.48      0.48      0.48      2257
           2       0.55      0.60      0.57      2257
           3       0.68      0.81      0.74      2257
           4       0.47      0.39      0.43      2257

    accuracy                           0.52     11285
   macro avg       0.51      0.52      0.51     11285
weighted avg       0.51      0.52      0.51     11285

Best Parameters  {'multi_class': 'auto', 'solver': 'saga'}
Best Score  0.6667065271549577
 
Confusion Matrix
[[ 719  522  512  182  322]
 [ 480 1081  275  187  234]
 [ 279  209 1354  138  277]
 [  88   90   90 1839  150]
 [ 423  353  235  361  885]]
 
accuracy: 0.5208684093929996, recall: 0.5208684093929996, precision: 0.5208684093929996, f1-score: 0.5208684093929996


e:\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:764: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)
e:\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:764: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_

e:\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:764: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)
e:\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:764: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_

e:\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:764: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)
e:\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:764: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_

e:\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:764: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)
e:\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:764: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_

Accuracy of Combinining word & Char n-gram 0.7369407075236671
              precision    recall  f1-score   support

           0       0.36      0.30      0.33      2257
           1       0.50      0.49      0.49      2257
           2       0.52      0.61      0.57      2257
           3       0.68      0.81      0.74      2257
           4       0.46      0.39      0.42      2257

    accuracy                           0.52     11285
   macro avg       0.51      0.52      0.51     11285
weighted avg       0.51      0.52      0.51     11285

Best Parameters  {'multi_class': 'auto', 'solver': 'saga'}
Best Score  0.6889187842551071
 
Confusion Matrix
[[ 671  477  569  207  333]
 [ 415 1110  300  196  236]
 [ 264  203 1387  131  272]
 [  76   91   94 1824  172]
 [ 421  353  292  318  873]]
 
accuracy: 0.5197164377492246, recall: 0.5197164377492246, precision: 0.5197164377492246, f1-score: 0.5197164377492246


In [171]:
from sklearn.tree import DecisionTreeClassifier
DTC1=DecisionTreeClassifier()
param={'criterion':('gini', 'entropy')}
print("Multi Nominal Bayes")
#trainer1111(MNB11,param, x_smote_Word,y_smote_Word, x_smote_Char,y_smote_Char, x_smote_Comb ,y_smote_Comb, x_smote_Word_Dev, y_smote_Word_Dev, x_smote_Char_Dev,y_smote_Char_Dev, x_smote_Comb_Dev, y_smote_Comb_Dev)
w_mnb,c_mnb,wc_mnb=trainer1111(DTC1,param, x_smote_Word,y_smote_Word, x_smote_Char,y_smote_Char, x_smote_Comb ,y_smote_Comb, x_smote_Word_Dev, y_smote_Word_Dev, x_smote_Char_Dev,y_smote_Char_Dev, x_smote_Comb_Dev, y_smote_Comb_Dev)

Multi Nominal Bayes
Accuracy of Word n-gram(1,1) : 0.21329198050509526
              precision    recall  f1-score   support

           0       0.44      0.01      0.02      2257
           1       0.09      0.00      0.00      2257
           2       0.65      0.04      0.08      2257
           3       0.20      0.99      0.34      2257
           4       0.69      0.03      0.06      2257

    accuracy                           0.21     11285
   macro avg       0.42      0.21      0.10     11285
weighted avg       0.42      0.21      0.10     11285

Best Parameters  {'criterion': 'entropy'}
Best Score  0.28045839561534625
 
Confusion Matrix
[[  18    3   14 2213    9]
 [   0    1    0 2256    0]
 [   9    1   96 2142    9]
 [   0    0   21 2224   12]
 [  14    6   16 2153   68]]
 
accuracy: 0.21329198050509526, recall: 0.21329198050509526, precision: 0.21329198050509526, f1-score: 0.21329198050509526
Accuracy of Char n-gram(1,5) : 0.45494018608772707
              precision    reca

In [74]:
from sklearn.ensemble import RandomForestClassifier
RFC11=RandomForestClassifier()
param={'n_estimators': [200, 400, 600, 800, 1000, 1200]}
print("RFC")
param={'kernel':('linear', 'rbf'), 'gamma':('auto', 'scale'), 'C':[0.9, 10, 20, 30]}
w_mnb,c_mnb,wc_mnb=trainer1(RFC11,param, x_smote_Word,y_smote_Word, x_smote_Char,y_smote_Char, x_smote_Comb ,y_smote_Comb, x_smote_Word_Dev, y_smote_Word_Dev, x_smote_Char_Dev,y_smote_Char_Dev, x_smote_Comb_Dev, y_smote_Comb_Dev)

RFC
Accuracy of Word n-gram(1,2) : 0.7606520517144463
              precision    recall  f1-score   support

           0       0.48      0.97      0.64      1779
           1       0.88      0.73      0.80      1779
           2       1.00      0.60      0.75      1779
           3       1.00      0.58      0.74      1779
           4       0.95      0.92      0.93      1779

    accuracy                           0.76      8895
   macro avg       0.86      0.76      0.77      8895
weighted avg       0.86      0.76      0.77      8895

Confusion Matrix
[[1727    1    0    0   51]
 [ 471 1300    0    0    8]
 [ 697    0 1071    0   11]
 [ 564  168    0 1036   11]
 [ 147    0    0    0 1632]]
 
Precision :  0.7606520517144463
Recall :  0.7606520517144463
F1 Score :  0.7606520517144463
Accuracy of Char n-gram(1,5) : 0.6197863968521642
              precision    recall  f1-score   support

           0       0.36      0.97      0.52      1779
           1       0.96      0.42      0.59   

In [ ]:
from sklearn.ensemble import RandomForestClassifier
RFC11=RandomForestClassifier()
param={'n_estimators': [200, 400, 600, 800, 1000, 1200]}
print("RFC")
#param={'kernel':('linear', 'rbf'), 'gamma':('auto', 'scale'), 'C':[0.9, 10, 20, 30]}
w_mnb,c_mnb,wc_mnb=trainer1111(RFC11,param, x_smote_Word,y_smote_Word, x_smote_Char,y_smote_Char, x_smote_Comb ,y_smote_Comb, x_smote_Word_Dev, y_smote_Word_Dev, x_smote_Char_Dev,y_smote_Char_Dev, x_smote_Comb_Dev, y_smote_Comb_Dev)

RFC


In [1]:
pip install inltk

Note: you may need to restart the kernel to use updated packages.


In [2]:
from inltk.inltk import setup
setup('hi')

In [1]:
pip install indic-nlp-library

Note: you may need to restart the kernel to use updated packages.


In [14]:
from indicnlp.transliterate.unicode_transliterate import UnicodeIndicTransliterator

# Input text "Today the weather is good. Sun is bright and there are no signs of rain. Hence we can play today."
input_text='ente mammookka ningal puliyalla oru simhama'

# Transliterate from Hindi to Telugu
print(UnicodeIndicTransliterator.transliterate(input_text,"en","ml"))

ente mammookka ningal puliyalla oru simhama


In [18]:
from indicnlp.transliterate.unicode_transliterate import ItransTransliterator

input_text=u'आज मौसम अच्छा है। इसलिए हम आज खेल सकते हैं!'

# Transliterate Hindi to Roman
print(ItransTransliterator.to_itrans(input_text, 'ta'))

आज मौसम अच्छा है। इसलिए हम आज खेल सकते हैं!


In [20]:
from indicnlp.transliterate.unicode_transliterate import ItransTransliterator

input_text = u'ஒன்றுமட்டுமல்லாது'
lang='ta'
input_text = ItransTransliterator.to_itrans(input_text,lang)
print(input_text)
#OUTPUT : .oऩRumaTTumallAtu

from indicnlp.transliterate.unicode_transliterate import ItransTransliterator
lang='ta'
x=ItransTransliterator.from_itrans(input_text,lang)
print(x) 
#OUTPUT :  ஒனறுமட்டுமல்லாது

ஒுமுமாு
ஒுமுமாு
